src/js/components/dropdown.js

```diff
@@ -0,0 +1,286 @@
+
+          /**
+           * If you were focused on the dropdown toggle
+           */
+          if (toggle && toggle !== null) {
+            // If you're focused on the toggle button and the menu is open.
+            if (toggle.getAttribute('aria-expanded') === 'true') {
+              const currentMenu = this._setUpMenu(this.menuElement);
+  
+              currentMenu.first.focus();
+            }
+  
+            this.openMenu(toggleButton, menuList);
+          }
+
+          /**
+           * Handle down arrow key when inside the open menu.
+           */
+
+          if (event.target.closest(this.menuAttribute) !== null) {
+            const currentMenu = this._setUpMenu(this.menuElement);
+
+            let currentIndex;
+
+            /**
+             * This keeps track of which button/focusable is focused in the open menu
+             */
+            for (let i = 0; i < currentMenu.all.length; i++) {
+              if (event.target == currentMenu.all[i]) {
+                currentIndex = i;
+              }
+            }
+
+            const nextItem = currentMenu.all[currentIndex + 1];
+
+            if (!nextItem) {
+              currentMenu.first.focus();
+
+              return;
+            }
+
+            nextItem.focus();
+          }
+  
+          break;
+        }
+  
+        case keyCodes.up: {
+          event.preventDefault();
+
+          /**
+           * Handle up arrow key when inside the open menu.
+           */
+          if (event.target.closest(this.menuAttribute) !== null) {
+            const currentMenu = this._setUpMenu(this.menuElement);
+
+            let currentIndex;
+
+            /**
+             * This keeps track of which button/focusable is focused in the open menu
+             */
+            for (let i = 0; i < currentMenu.all.length; i++) {
+              if (event.target == currentMenu.all[i]) {
+                currentIndex = i;
+              }
+            }
+
+            const previousItem = currentMenu.all[currentIndex - 1];
+
+            if (!previousItem && currentMenu.last !== undefined) {
+              currentMenu.last.focus();
+
+              return;
+            }
+
+            previousItem.focus();
+          }
+  
+          break;
+        }
+  
+        case keyCodes.escape: {
+          // If there's an open menu, close it.
+          if (this.activeDropdown) {
+            this.closeMenu(this.toggleElement, this.menuElement);
+          }
+
+          if (this.toggleElement && this.toggleElement !== null) {
+            this.toggleElement.focus();
+          }
+
+          /**
+           * Resets the state variables so as not to interfere with other
+           * Escape key handlers/interactions
+           */
+          this.activeDropdown = null;
+
+          break;
+        }
+  
+        case keyCodes.tab: {
+          /**
+           * Handle tab key when inside the open menu.
+           */
+          if (event.target.closest(this.menuAttribute) !== null) {
```

Comment:
I'm seeing a console error here in IE11. Not super urgent right now, but we'll need to look into this and figure out how to stop that from happening. It doesn't break anything, but we'll need to get if figured out eventually. 👍